### ISSUE TITLE
Expensify Card - Admin can go through Add card flow when bank account is not added

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#50494]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.46-4
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** applausetester+pso@applause.expensifail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
Precondition:
- Expensify Card feature is enabled.
- Bank account is not added.

1. Go to staging.new.expensify.com
2. Go to workspace settings > Members.
3. Click on any member.
4. Click Add card.
5. Go through the flow.

## Expected Result:
Admin should not be able to go through Add card flow when bank account is not added.

## Actual Result:
Admin can go through Add card flow when bank account is not added. The card is not added in the end.

## Workaround:
Unknown

## Platforms:

- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x ] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/02c6096a-ba90-488f-a00a-8f1956711984
 
</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by IuliiaHerets** [comment]
@CortneyOfstad  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by ChavdaSachin** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Admin can go through Add card flow when bank account is not added

### What is the root cause of that problem?
There are a total of three check we perform before starting `issueNewExpensifyCard` flow, And based on these checks we redirect user to appropriate page to complete prior setup.
1. `isCurrencySupportedForDirectReimbursement` which we perform here, and ask user to switch to supported currency aka. USD if check fails.
https://github.com/Expensify/App/blob/2bed8d7018ca3ba3ea073eb64afc6ebd17d93d1a/src/pages/workspace/expensifyCard/WorkspaceExpensifyCardPageEmptyState.tsx#L95
2. Whether admin has added any bank account. If not we redirect user to add bank account here.
https://github.com/Expensify/App/blob/2bed8d7018ca3ba3ea073eb64afc6ebd17d93d1a/src/pages/workspace/expensifyCard/WorkspaceExpensifyCardPageEmptyState.tsx#L61-L62
3. Admin has set default reimbursement account for workspace.If not we redirect admin to select default reimbursement account for the workspace. For which we redirect user to `WorkspaceExpensifyCardBankAccounts` page. 
https://github.com/Expensify/App/blob/2bed8d7018ca3ba3ea073eb64afc6ebd17d93d1a/src/pages/workspace/expensifyCard/WorkspaceExpensifyCardPageEmptyState.tsx#L63-L65
These checks are only performed from component `WorkspaceExpensifyCardPageEmptyState`

And we do not perform all of these check when we start `issueNewExpensifyCard` flow rom `workspaceMemberDetailsPage` here.
https://github.com/Expensify/App/blob/a4273274a74e106ed6c6cf711ebea5a8e995c0be/src/pages/workspace/members/WorkspaceMemberDetailsPage.tsx#L166-L181

### What changes do you think we should make in order to solve the problem?
We have a total of 2 workflow flow for issuing New Expensify Card.
1. From `workspaceExpensifyCardPage` which is working perfectly as of now.
2. Issue card from `workspaceMemberDetailsPage`

I propose to migrate all of the above three checks mentioned in root cause to [IssueNewCardPage.tsx](https://github.com/Expensify/App/blob/81e914a194e8a769b326523793e1bafede0f7b8f/src/pages/workspace/expensifyCard/issueNew/IssueNewCardPage.tsx).

To keep proposal neat, not adding all code instead here's the [test branch](https://github.com/ChavdaSachin/App/tree/fix-50494)

> [!Important]
Reason why I am not suggesting to perform those checks of `workspaceMemberDetailsPage` and rather to migrate to `IssueNewCardPage` is...
When Company cards are enabled for the workspace along with Expensify Cards - when we click add new card from workspaceMemberDetailsPage` we list all Company cards along with Expensify Card and all of the above checks are meant only for Expensify Card and we don't want to stop admin from issuing company cards without fulfilling those checks. So rather we should perform those checks only when admin selects Expensify Card

And if there are any minor issues in this we could address it during PR phase.

### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by ChavdaSachin** [comment]
Above proposal is the overall idea, there are a few navigation related minor changes which we could address during PR phase.

##Result

https://github.com/user-attachments/assets/da39bdf8-d65b-41bf-a828-7188359d1dc6

**Comment by CortneyOfstad** [comment]
@IuliiaHerets — I'm a bit confused by the pre-requisites. Within Expensify, Expensify Cards are not enabled until a bank account is in place for customers, so there is no situation where they would be able to even be prompted to add cards without a business bank account being added **AND** verified within Expensify.

**Comment by ChavdaSachin** [comment]
@CortneyOfstad Are you suggesting we should prevent user from enabling Expensify Cards feature in the first place if user has not configured bank account yet ?

Summary of behavior on ND,
1. User is able to enable enable Expensify Cards on features page even though no bank account is linked.
2. When user tries to issue a new Expensify Card from WorkspaceExpensifyCardsPage - user is redirected to add bank account page if no bank account is linked to user account, if user already have linked a bank account but default reimbursement account is not yet set for the workspace, then user is redirected to set default account for workspace reimbursement.
3. When user tries to issue new Expensify Card from WorkspaceMemberDetailsPage , user is directly sent to issueNewExpensifyCard flow.

As of now my understanding for this issue is to refactor Point 3 here, and make sure behavior for point 3 is similar to point 2 here.

@CortneyOfstad You can ignore rest of the message if you are not checking proposals as of now.

## Addition to my proposal

- Are you suggesting we should prevent user from enabling Expensify Cards feature in the first place if user has not configured bank account yet ?
<details><summary><h3>If yes</h3>, then here's an addition to my <a href="https://github.com/Expensify/App/issues/50494#issuecomment-2402758761" >Proposal</a></summary>

1. For those bank account related checks I mentioned in the proposal, rather then migrating them to issueNewExpensifyCard flow, we should use those checks before user enables Expensify Cards feature itself.

2. When user tries to toggle Expensify Cards page first check if bank account is not linked, redirect user to link bank account and upon successfully linking bank account let user select default reimbursement account for the workspace.  Upon success enable Expensify Cards feature.

3. When user tries to enable Expensify Cards feature, and bank account is linked but default reimbursement account is not set for the workspace - redirect user to select default reimbursement account for the workspace. Upon success enable Expensify Cards feature.
4. We need to configure BE to return an error.
</details>
